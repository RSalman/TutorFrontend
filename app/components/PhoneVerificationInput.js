import React, { Component } from 'react';
import { StyleSheet, Text, TextInput, View } from 'react-native';
import { Actions } from 'react-native-router-flux';
import { bindActionCreators } from 'redux';
import { connect } from 'react-redux';
import { verifyCode } from '../actions/signup';

const CODE_LENGTH = 4;

class PhoneVerificationInput extends Component {
  constructor(props) {
    super(props);
    this.state = { isCodeValid: false };
  }

  componentWillReceiveProps(nextProps) {
    if (nextProps.verified)
      Actions.muraad({ type: 'reset' }); // reset so user can't go back
  }

  verify(code) {
    const valid = code && code.length === CODE_LENGTH;
    this.setState({ isCodeValid: valid });
    if (valid)
      this.props.verifyCode(code);
  }

  render() {
    return (
      <View style={styles.container}>
        <View style={styles.headerContainer}>
          <Text style={styles.headerText}>Please enter the code {'\n'} sent to your phone number</Text>
        </View>
        <TextInput
          style={styles.input}
          autoFocus
          onChangeText={(code) => this.verify(code)}
          disabled={this.props.verifying}
          maxLength={CODE_LENGTH}
          keyboardType="numeric"
        />
        <View style={styles.textContainer}>
          <Text style={styles.text}>
            {(!this.props.verifying && !this.props.verified && this.state.isCodeValid) ?
                'Incorrect code entered! Please try again.' : ''
            }
          </Text>
        </View>
      </View>
    );
  }
}

const failureTextColor = 'red';
const inputBorderColor = 'gray';
const styles = StyleSheet.create({
  container: {
    flex: 1,
    justifyContent: 'center',
  },
  headerContainer: {
    alignSelf: 'center',
    marginBottom: 15,
  },
  headerText: {
    fontSize: 20,
    textAlign: 'center',
  },
  input: {
    alignSelf: 'center',
    height: 100,
    width: 230,
    fontSize: 85,
    padding: 5,
    textAlign: 'center',
    borderColor: inputBorderColor,
    borderWidth: 1
  },
  textContainer: {
    marginTop: 15,
    height: 100,
    width: null,
  },
  text: {
    fontSize: 12,
    textAlign: 'center',
    color: failureTextColor
  }
});

const mapStateToProps = (state) => {
  return {
    verifying: state.signup.phoneVerifying,
    verified: state.signup.phoneVerified
  };
};

const mapDispatchToProps = (dispatch) => bindActionCreators({ verifyCode }, dispatch);

export default connect(mapStateToProps, mapDispatchToProps)(PhoneVerificationInput);
