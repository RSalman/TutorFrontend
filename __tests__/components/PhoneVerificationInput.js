import React from 'react';
import renderer from 'react-test-renderer';
import configureStore from 'redux-mock-store';
import thunk from 'redux-thunk';
import PhoneVerificationInput from '../../app/components/PhoneVerificationInput';

const middlewares = [thunk];
const mockStore = configureStore(middlewares);

function setup(verified, verifying) {
  const store = mockStore({
    signup: {
      phoneVerified: verified,
      phoneVerifying: verifying
    }
  });
  const props = {
    verifyCode: jest.fn(),
    store
  };
  return props;
}

describe('<PhoneVerificationInput />', () => {
  it('should disable input when verifying', () => {
    const props = setup(false, true);
    const tree = renderer.create(
      <PhoneVerificationInput {...props} />
    ).toJSON();

    expect(tree).toMatchSnapshot();
  });

  it('should show no message when invalid code entered', () => {
    const props = setup(false, false);
    const tree = renderer.create(
      <PhoneVerificationInput {...props} />
    ).toJSON();

    expect(tree).toMatchSnapshot();
  });
});
