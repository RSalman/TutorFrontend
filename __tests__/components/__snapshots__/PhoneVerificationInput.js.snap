exports[`<PhoneVerificationInput /> should disable input when verifying 1`] = `
<View
  style={
    Object {
      "flex": 1,
      "justifyContent": "center",
      "marginTop": 60,
    }
  }>
  <View
    style={
      Object {
        "alignSelf": "center",
        "marginBottom": 15,
      }
    }>
    <Text
      accessible={true}
      allowFontScaling={true}
      ellipsizeMode="tail"
      style={
        Array [
          Object {
            "fontFamily": "HelveticaNeue-Thin",
          },
          Object {
            "fontSize": 20,
            "textAlign": "center",
          },
        ]
      }>
      Please enter the code 
       sent to your phone number
    </Text>
  </View>
  <TextInput
    autoFocus={true}
    disabled={true}
    keyboardType="numeric"
    maxLength={4}
    onChangeText={[Function]}
    style={
      Object {
        "alignSelf": "center",
        "borderColor": "gray",
        "borderWidth": 1,
        "fontSize": 85,
        "height": 100,
        "padding": 5,
        "textAlign": "center",
        "width": 230,
      }
    } />
  <View
    style={
      Object {
        "height": 100,
        "marginTop": 15,
        "width": null,
      }
    }>
    <Text
      accessible={true}
      allowFontScaling={true}
      ellipsizeMode="tail"
      style={
        Array [
          Object {
            "fontFamily": "HelveticaNeue-Thin",
          },
          Object {
            "color": "red",
            "fontSize": 12,
            "textAlign": "center",
          },
        ]
      }>
      
    </Text>
  </View>
</View>
`;

exports[`<PhoneVerificationInput /> should show no message when invalid code entered 1`] = `
<View
  style={
    Object {
      "flex": 1,
      "justifyContent": "center",
      "marginTop": 60,
    }
  }>
  <View
    style={
      Object {
        "alignSelf": "center",
        "marginBottom": 15,
      }
    }>
    <Text
      accessible={true}
      allowFontScaling={true}
      ellipsizeMode="tail"
      style={
        Array [
          Object {
            "fontFamily": "HelveticaNeue-Thin",
          },
          Object {
            "fontSize": 20,
            "textAlign": "center",
          },
        ]
      }>
      Please enter the code 
       sent to your phone number
    </Text>
  </View>
  <TextInput
    autoFocus={true}
    disabled={false}
    keyboardType="numeric"
    maxLength={4}
    onChangeText={[Function]}
    style={
      Object {
        "alignSelf": "center",
        "borderColor": "gray",
        "borderWidth": 1,
        "fontSize": 85,
        "height": 100,
        "padding": 5,
        "textAlign": "center",
        "width": 230,
      }
    } />
  <View
    style={
      Object {
        "height": 100,
        "marginTop": 15,
        "width": null,
      }
    }>
    <Text
      accessible={true}
      allowFontScaling={true}
      ellipsizeMode="tail"
      style={
        Array [
          Object {
            "fontFamily": "HelveticaNeue-Thin",
          },
          Object {
            "color": "red",
            "fontSize": 12,
            "textAlign": "center",
          },
        ]
      }>
      
    </Text>
  </View>
</View>
`;
